### DISCUSSION TITLE
Lookup a list if item exists, in a separate file (json/yaml) that is dynamically generated.

### DISCUSSION THREAD

**Discussion by Roohiya** [cue-lang/cue#1574] [Q&A]
I would like to validate a value of a field in a struct against a separately generated list of items. From the current documentation there's no info on importing a pre-generated field in a cue file.

json file:
```
[item_1, item_2, ...]
```

cue file to import it and loop through these values to test whether cue data is valid. Furthermore, it will be interesting if CUE can allow nested looping. Understand this a data language and not a full-fledged programming language, but wanted to see if I can push this tool for more.

**Comment by jlongtine** [comment]
@Roohiya There's no way (currently... look here for proposals) to import a non-cue file into the cue lattice. 

You _can_ include a number of filetypes in a cue evaluation, however. See `cue filetypes` for more information on that (and ping me if you have questions!). 

Also, what do you mean by nested looping? This, for instance, is valid CUE:

```cue
a: [1,2,3,4]
b: {
  for i in a {
    for j in a {
      "\(i)+\(j)": i+j
    }
  }
}
```

(play with it here: https://tip.cuelang.org/play/?id=Dn-uaLP4Tx1#cue@export@cue)

I hope that helps!

**Comment by Roohiya** [comment]
Hey @jlongtine, thanks for your reply! Understand the nested loop 👌

As for the other issue, is it possible to a generate/create a cue file, with a dynamic array of items (most probably populated by an API call)? And have this file imported in a .cue (schema) file. For example

1. Dynamically generated file (API call to get a list) - **list.cue**
Note: I realise that cue is hermetic and does not allow interaction with the outside world. But hoping there's a way to check against a super long list of items which can't be hardcoded in the cue schema. Can CUE handle this?

```
[
   {name: "item_1"},
   {name: "item_2"}
]
```

2. In **validate.cue** file, import the above list file:
```
import list


#object: {
   name: string & <if name exists in the imported list>
}

```

Is this possible though?

**Reply by Roohiya** [reply]
Figured it out - quite trivial

We can separately generate a json file which can be converted to a cue package via
```cue import -p <package-name> <json-file>.json```

Subsequently this can be imported into a cue schema file for validation
